packages/shared/src/components/squads/SquadHeaderBar.tsx

```diff
@@ -29,33 +36,17 @@ import { UserIntegrationType } from '../../graphql/integrations';
 import { useAuthContext } from '../../contexts/AuthContext';
 import { ProfileImageSize } from '../ProfilePicture';
 
-export function SquadHeaderBar({
+type SquadBarButtonProps<T extends AllowedTags> = Pick<
+  Partial<ButtonProps<T>>,
+  'onClick' | 'disabled' | 'icon'
+> &
+  Pick<SquadMemberShortListProps, 'squad'>;
+
+const SquadSlackButton = <T extends AllowedTags>({
   squad,
```

Comment:
Splitted from main component so each component is easy to understand and have separated logic

Reply:
Can you specify which block to be considered as the actual change?

Reply:
Only `SquadModerationButton` is the new component inside `SquadHeaderBar` ✔️